// ariatemplates/build/target/bootstrap/aria/touch/widgets/ButtonCfg.js
var Aria = require("../../Aria");
var ariaCoreJsonTypes = require("../../core/JsonTypes");
var ariaHtmlBeansElementCfg = require("../../html/beans/ElementCfg");
var ariaWidgetLibsCommonBeans = require("../../widgetLibs/CommonBeans");

module.exports = Aria.beanDefinitions({
    $package : "aria.touch.widgets.ButtonCfg",
    $description : "Configuration for Button widget.",
    $namespaces : {
        "json" : ariaCoreJsonTypes,
        "base" : ariaHtmlBeansElementCfg,
        "common" : ariaWidgetLibsCommonBeans
    },
    $beans : {
        "Properties" : {
            $type : "base:Properties",
            $description : "Properties of a Button widget.",
            $properties : {
                "on" : {
                    $type : "base:Properties.$properties.on",
                    $properties : {
                        "type" : {
                            $type : "common:Callback",
                            $description : "Callback called when the user interacts with the button."
                        }
                    }
                },
                "isLink" : {
                    $type : "json:Boolean",
                    $description : "Whether the button is a link, which means a different highlighting pattern.",
                    $default : false
                },
                "delay" : {
                    $type : "json:Integer",
                    $description : "delay between and tapstart event and highlighting of the link or button in milliseconds."
                },
                "tagName" : {
                    $type : "base:Properties.$properties.tagName",
                    $description : "Tag to be used for Button markup."
                }
            }
        }
    }
});
