// ariatemplates/build/target/bootstrap/aria/utils/Beans.js
var Aria = require("../Aria");
var ariaCoreJsonTypes = require("../core/JsonTypes");
var ariaCoreEnvironmentEnvironmentBaseCfgBeans = require("../core/environment/EnvironmentBaseCfgBeans");


/**
 * @class aria.utils.Beans
 */
module.exports = Aria.beanDefinitions({
    $package : "aria.utils.Beans",
    $description : "",
    $namespaces : {
        "json" : ariaCoreJsonTypes,
        "environmentBase" : ariaCoreEnvironmentEnvironmentBaseCfgBeans
    },
    $beans : {
        "options" : {
            $type : "json:Object",
            $description : "Configuration bean used by the interpreter of Date.js",
            $properties : {
                "referenceDate" : {
                    $type : "json:Date",
                    $description : "Reference date from which value can be calculated."
                },
                "inputPattern" : {
                    $type : "environmentBase:inputFormatTypes",
                    $description : "Date pattern used to match user input to convert it in a Javascript valid date."
                }
            }
        }
    }
});
