// ariatemplates/build/target/bootstrap/aria/resources/handlers/LCRangeResourceHandlerBean.js
var Aria = require("../../Aria");
var ariaResourcesHandlersLCResourcesHandlerBean = require("./LCResourcesHandlerBean");
var ariaCoreJsonTypes = require("../../core/JsonTypes");

/**
 * Definition of the suggestions used in the MultiAutoComplete Handler
 * @class aria.resources.handlers.MultiAutoCompleteHandlerBean
 */
module.exports = Aria.beanDefinitions({
    $package : "aria.resources.handlers.LCRangeResourceHandlerBean",
    $description : "Definition of the suggestions used in the MultiAutoComplete resource handler",
    $namespaces : {
        "base" : ariaResourcesHandlersLCResourcesHandlerBean,
        "json" : ariaCoreJsonTypes
    },
    $beans : {
        "Configuration" : {
            $type : "base:Configuration",
            $description : "Configuration Object for Suggestions with range of values",
            $restricted : false,
            $properties : {
                "allowRangeValues" : {
                    $type : "json:Boolean",
                    $description : "To add range of values",
                    $default : false
                }
            }
        }
    }
});
