// ariatemplates/build/target/bootstrap/aria/tools/contextual/environment/ContextualMenu.js
var Aria = require("../../../Aria");
require("./ContextualMenuCfgBeans");
var ariaCoreEnvironmentEnvironmentBase = require("../../../core/environment/EnvironmentBase");


/**
 * Contains getters for the ContextualMenu environment.
 * @class aria.tools.contextual.environment.ContextualMenu
 * @extends aria.core.environment.EnvironmentBase
 * @singleton
 */
module.exports = Aria.classDefinition({
    $classpath : "aria.tools.contextual.environment.ContextualMenu",
    $extends : ariaCoreEnvironmentEnvironmentBase,
    $singleton : true,
    $prototype : {
        /**
         * Classpath of the bean which allows to validate the part of the environment managed by this class.
         * @type String
         */
        _cfgPackage : "aria.tools.contextual.environment.ContextualMenuCfgBeans.AppCfg",

        /**
         * Returns the contextual menu settings
         * @public
         * @return {aria.core.environment.environment.EnvironmentBaseCfgBeans:AppCfg}
         */
        getContextualMenu : function () {
            return this.checkApplicationSettings("contextualMenu");
        }
    }
});
