// ariatemplates/build/target/bootstrap/aria/utils/environment/Number.js
var Aria = require("../../Aria");
require("./NumberCfgBeans");
var ariaCoreEnvironmentEnvironmentBase = require("../../core/environment/EnvironmentBase");


/**
 * Contains getters for the Number environment.
 */
module.exports = Aria.classDefinition({
    $classpath : "aria.utils.environment.Number",
    $extends : ariaCoreEnvironmentEnvironmentBase,
    $singleton : true,
    $prototype : {
        /**
         * Classpath of the bean which allows to validate the part of the environment managed by this class.
         * @type String
         */
        _cfgPackage : "aria.utils.environment.NumberCfgBeans.AppCfg",

        /**
         * Return currency formats
         * @return {aria.utils.environment.NumberCfgBeans:CurrencyFormatsCfg}
         */
        getCurrencyFormats : function () {
            return this.checkApplicationSettings("currencyFormats");
        },

        /**
         * Return decimal format symbols
         * @return {aria.utils.environment.NumberCfgBeans:DecimalFormatSymbols}
         */
        getDecimalFormatSymbols : function () {
            return this.checkApplicationSettings("decimalFormatSymbols");
        }
    }
});
