// ariatemplates/build/target/bootstrap/aria/templates/CmlClassGenerator.js
var Aria = require("../Aria");
var ariaTemplatesCSSParser = require("./CSSParser");
var ariaTemplatesCSSClassGenerator = require("./CSSClassGenerator");
var ariaTemplatesClassGenerator = require("./ClassGenerator");


/**
 * Generate the class definition for a CSS Template library
 * @class aria.templates.CmlClassGenerator
 */
module.exports = Aria.classDefinition({
    $classpath : 'aria.templates.CmlClassGenerator',
    $extends : ariaTemplatesClassGenerator,
    $singleton : true,
    $constructor : function () {
        this.$ClassGenerator.constructor.call(this);

        // Load the Template specific statements
        this._loadStatements(["CSSLibrary"]);

        // Redefine the protected parser
        this._parser = ariaTemplatesCSSParser;

        // Redefine the class used as the parent for templates which do not inherit from any other template
        this._superClass = "aria.templates.CSSTemplate";

        this._classType = "CML";
        this._rootStatement = "CSSLibrary";
        this._templateParamBean = "aria.templates.CfgBeans.CSSLibraryCfg";
    },
    $prototype : {
        /**
         * Write to the current block of the class writer the $init method which is used both to import the script
         * prototype (if any) and to handle csslibs inheritance.
         * @param {aria.templates.ClassWriter} out
         * @protected
         */
        _writeClassInit : function (out) {
            ariaTemplatesCSSClassGenerator._writeClassInit.call(this, out);
        }
    }
});
