// ariatemplates/build/target/bootstrap/aria/html/beans/AutoCompleteCfg.js
var Aria = require("../../Aria");
var ariaCoreJsonTypes = require("../../core/JsonTypes");
var ariaHtmlBeansTextInputCfg = require("./TextInputCfg");


module.exports = Aria.beanDefinitions({
    $package : "aria.html.beans.AutoCompleteCfg",
    $description : "Configuration for AutoComplete widget.",
    $namespaces : {
        "json" : ariaCoreJsonTypes,
        "input" : ariaHtmlBeansTextInputCfg
    },
    $beans : {
        "Properties" : {
            $type : "input:Properties",
            $description : "Properties of an AutoComplete widget.",
            $properties : {
                "bind" : {
                    $type : "input:Properties.bind",
                    $properties : {
                        "suggestions" : {
                            $type : "json:Array",
                            $description : "List of suggestions taken from the Resources Handler",
                            $contentType : {
                                $type : "json:Object",
                                $description : "Suggestion"
                            },
                            $default : []
                        }
                    }
                }
            }
        }
    }
});
