// ariatemplates/build/target/bootstrap/aria/html/beans/CheckBoxCfg.js
var Aria = require("../../Aria");
var ariaHtmlBeansInputElementCfg = require("./InputElementCfg");
var ariaWidgetLibsCommonBeans = require("../../widgetLibs/CommonBeans");


module.exports = Aria.beanDefinitions({
    $package : "aria.html.beans.CheckBoxCfg",
    $description : "Configuration for CheckBox widget.",
    $namespaces : {
        "base" : ariaHtmlBeansInputElementCfg,
        "common" : ariaWidgetLibsCommonBeans
    },
    $beans : {
        "Properties" : {
            $type : "base:Properties",
            $description : "Properties of a CheckBox widget.",
            $properties : {
                "bind" : {
                    $type : "base:Properties.bind",
                    $properties : {
                        "checked" : {
                            $type : "common:BindingRef",
                            $description : "Bi-directional binding. The text input's checked property is set in the bound object."
                        }
                    }
                }
            }
        }
    }
});
