// ariatemplates/build/target/bootstrap/aria/html/beans/InputElementCfg.js
var Aria = require("../../Aria");
var ariaHtmlBeansElementCfg = require("./ElementCfg");
var ariaWidgetLibsCommonBeans = require("../../widgetLibs/CommonBeans");


module.exports = Aria.beanDefinitions({
    $package : "aria.html.beans.InputElementCfg",
    $description : "Configuration for the generic InputElement.",
    $namespaces : {
        "base" : ariaHtmlBeansElementCfg,
        "common" : ariaWidgetLibsCommonBeans
    },
    $beans : {
        "Properties" : {
            $type : "base:Properties",
            $description : "Properties of an InputElement.",
            $properties : {
                "bind" : {
                    $type : "base:Properties.bind",
                    $properties : {
                        "disabled" : {
                            $type : "common:BindingRef",
                            $description : "Binding for the disabled attribute of the input element."
                        }
                    }
                }
            }
        }
    }
});
