// ariatemplates/build/target/bootstrap/aria/html/beans/RadioButtonCfg.js
var Aria = require("../../Aria");
var ariaCoreJsonTypes = require("../../core/JsonTypes");
var ariaHtmlBeansInputElementCfg = require("./InputElementCfg");
var ariaWidgetLibsCommonBeans = require("../../widgetLibs/CommonBeans");


module.exports = Aria.beanDefinitions({
    $package : "aria.html.beans.RadioButtonCfg",
    $description : "Configuration for RadioButton widget.",
    $namespaces : {
        "json" : ariaCoreJsonTypes,
        "base" : ariaHtmlBeansInputElementCfg,
        "common" : ariaWidgetLibsCommonBeans
    },
    $beans : {
        "Properties" : {
            $type : "base:Properties",
            $description : "Properties of a RadioButton widget.",
            $properties : {
                "value" : {
                    $type : "json:String",
                    $description : "The value associated with the radio button."
                },
                "bind" : {
                    $type : "base:Properties.$properties.bind",
                    $properties : {
                        "selectedValue" : {
                            $type : "common:BindingRef",
                            $description : "Bi-directional binding."
                        }
                    }
                }
            }
        }
    }
});
