// ariatemplates/build/target/bootstrap/aria/widgetLibs/environment/WidgetLibsSettingsCfgBeans.js
var Aria = require("../../Aria");
var ariaCoreJsonTypes = require("../../core/JsonTypes");


/**
 * Bean definition containing default settings for the WidgetLibs environment.
 */
module.exports = Aria.beanDefinitions({
    $package : "aria.widgetLibs.environment.WidgetLibsSettingsCfgBeans",
    $namespaces : {
        "json" : ariaCoreJsonTypes
    },
    $description : "",
    $beans : {
        "AppCfg" : {
            $type : "json:Object",
            $description : "",
            $restricted : false,
            $properties : {
                "defaultWidgetLibs" : {
                    $type : "json:Map",
                    $description : "Widget libraries to be available by default in all templates. The key in the map is the prefix used inside the template to refer to that widget library. The value is the classpath of the library. The settings in the environment can be overridden in templates if the same key is used. See also the $wlibs property of the {Template} statement in aria.templates.CfgBeans.TemplateCfg.$wlibs.",
                    $contentType : {
                        $type : "json:PackageName",
                        $description : "Classpath of the widget library."
                    },
                    $default : {
                        "aria" : "aria.widgets.AriaLib"
                    }
                }
            }
        }
    }
});
